utilities/include/vec3.hpp

```diff
@@ -1,212 +1,63 @@
 #ifndef VEC3_HPP
 #define VEC3_HPP
 
+#include <cmath>
```

Comment:
We should actually overload `operator<<` for `vec3` to print 3 elements instead of 4.

Reply:
I actually thought of that. But then there's a lot of things we have to do to override inherited function. That's i initially included the `iostream` header but removed it afterwards. We may look into that but there's a lot of stuff breaking in builds. We might want to look into that first

Reply:
> I actually thought of that. But then there's a lot of things we have to do to override inherited function.

I don't think that overriding this function is too difficult. (See a dummy implementation here: https://www.godbolt.org/z/fd1aMEbbP)

> We may look into that but there's a lot of stuff breaking in builds. We might want to look into that first

You are right about that; that is a very real concern. Since Mahesh is near completion in his Scene class PR, I am assigning the task of making our project compile, to him.